go/mysql/filepos_gtid.go

```
@@ -133,6 +133,16 @@ func (gtid filePosGTID) AddGTID(other GTID) GTIDSet {
 	return filePosOther
 }
 
+// Union implements GTIDSet.Union().
+func (gtid filePosGTID) Union(other GTIDSet) GTIDSet {
+	filePosOther, ok := other.(filePosGTID)
+	if !ok || gtid.Contains(other) {
```

Sure you want to handle the !ok case by failing silently?

---

My thought was that if other is not a filePosGTID, then the union is the receiver.

---

@deepthi @enisoc thoughts here? I'm not opposed to allowing it to panic. I was somewhat following the pattern of other methods on `filePosGTID`.

---

I vote for keeping it consistent with the rest of the functions.

---

Ok, then, in that case, I think this is ready for another round of reviews.